core/runtime/src/main/java/io/quarkus/runtime/annotations/RunOnWorkerThread.java

```diff
@@ -0,0 +1,22 @@
+package io.quarkus.runtime.annotations;
+
+import static java.lang.annotation.ElementType.*;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import javax.inject.Qualifier;
+
+/**
+ * Qualifier used to indicate that the annotated element should not be executed on the IO thread but on
+ * a worker thread.
+ * This qualifier is used to allow <em>blocking</em> processing in a reactive application.
+ * For example, use this annotation if your reactive route or gRPC service need to use blocking APIs such
+ * as <em>blocking</em> database access.
+ */
+@Qualifier
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ TYPE, FIELD, METHOD })
+public @interface RunOnWorkerThread {
```

Comment:
Hm, I don't like the name. It probably shouldn't contain the "thread" and "worker" is an internal Vertx-specific name. Why not just `@Blocking`?

Reply:
I used @Blocking in https://github.com/quarkusio/quarkus/issues/12077, but @stuartwdouglas didn't like it :-D

Reply:
I don't see any discussion there ;-)

Reply:
Agree on "thread" name, as I also presume Loom would change that meaning.

Maybe `WorkOffload` if `Blocking` isn't liked

Reply:
VirtualThread are still Thread, they implement the same interface :-) But, I see what mean. 

One problem with @Blocking is the negative "meaning".

Reply:
For me the only blocker to approving the PR is this name

Reply:
I still don't understand why it's a CDI qualifier...

Reply:
Hm, I like the reference to "computation offloading". `@Offload`, `@Offloaded`... but it's probably not very intuitive...

Reply:
BTW if the name is meant to stay (which I also disagree), it should read `@RunsOnWorkerThread`, no?

Reply:
I've listed the proposals and wait for your vote :-D

Reply:
I think we need to be able to specify both blocking and non-blocking semantics, because it will depend on the default. e.g. reactive messaging will be non-blocking by default, but if we have support for non-blocking JAX-RS it may still default to blocking by default.

In terms of name what about `@RunsOn(WORKER_THREAD)`, `@RunsOn(IO_THREAD)` ?

Reply:
Anyone who's ever done web programming knows the difference between IO and WORKER pools. If they don't we need to educate them.

Reply:
I mean to say that using those terms is fine.

Reply:
> Anyone who's ever done web programming knows the difference between IO and WORKER pools. If they don't we need to educate them.

Well, I think that most of the Java EE programmers never heard about it ;-)

Reply:
I've seen many people use the term "IO thread" to actually mean worker thread. IMHO, that term should be banned for eternity. Either it's an event loop thread, or a worker thread.

(Stupid question: event loop threads don't do any IO. Why are they called "IO threads"?)

Reply:
Event loop threads do all the IO. Blocking IO based on Netty relies on the event loop thread delivering the data then waking up the blocking thread.